### ISSUE TITLE
`string-explode` does not seem to have an inverse

### ISSUE THREAD

**Issue by shriram** [brownplt/pyret-lang#836]
It seems odd to not have a `string-implode` or some other such function that can take the output of `string-explode`, perhaps processed slightly, and turn the result back into a string. I expect it would be something like this, but without the quadratic from string concatenation.
```
  fun string-implode(ss :: List<String>) -> String:
    for fold(base from "", s from ss):
      base + s
    end
  end
```

**Comment by jpolitz** [comment]
```
>string-explode("abc").join-str("")
"abc"
```

Does this suit your needs?

**Comment by shriram** [comment]
Yep. Naming/documentation needs improvement, then. (e.g., the two functions ought to refer to one another.) Incidentally, `join-str` doesn't seem to be mentioned in http://www.pyret.org/docs//latest/strings.html

**Comment by sorawee** [comment]
I agree with Shriram. Moreover, `join-str` should be in string rather than lists. This is one thing I really like about Python. `.join` is a string method, not list.

**Comment by blerner** [comment]
Mrph, that's something I don't like as much about Python, and it does _not_ carry over into Pyret well.  Strings (and numbers and booleans) are primitives not objects, so they can't have methods.  So it's not `",".join([list: 1, 2, 3, 4])` (to use Pyret syntax), but `[list: 1, 2, 3, 4].join-str(",")`.  If I were looking up documentation about "how do I combine a list into one string?", I don't think I'd look in the strings documentation first, but rather in the lists documentation.  

More importantly than "does `join-str` belong in lists.html or in strings.html?", I think this just highlights that we need an index or glossary page somehow.  I don't know yet what it should look like, exactly, but I know we need one.

**Comment by shriram** [comment]
I don't think it's meaningful to talk about what's “natural”, especially when this bug report began with me not finding something that was there because I was looking in precisely the place you said you wouldn't look. In any type-driven completion tool, there's value to having both input-driven and output-driven indexing: sometimes have have the pieces and need to figure out how to combine them, sometimes you have the target and need to figure out how to get there.

A search engine that let you specify "stuff on the input side, stuff on the output side" and let you search by that would be pretty neat.
